### PR TITLE
102 patternmarkers all returning empty lists

### PR DESCRIPTION
**Pull request by dimalvovs**
- add a test that fails on empty patterMarkers
- make it pass by rewriting the method for threshold='all'

### CODE REVIEW

**Comment by dimalvovs** [comment]
@kwoyshner can you review please?

**Comment by kwoyshner** [comment]
I tried reinstalling from this branch and was able to get patternMarkers with 'all' now, thank you!

**Comment by dimalvovs** [comment]
@kwoyshner do you want to have a look at `threshold = 'cut'`? I've updated it too. This part of code is poorly documented so I was not sure what that mode should do but there was a dimension mismatch. `simplicityGENES` function needs to be inspected for functionality.

File: R/methods-CogapsResult.R
Comment (jmitchell81) [inline R/methods-CogapsResult.R:1]: I'd suggest defining the simplicityGENES() function for ranking pattern markers scores outside of the definition of patternMarkers() for the sake of readability and documentation. Otherwise, LGTM.

**Review by drbergman — COMMENTED** [review]
I am unqualified to comment on how these changes affect the integrity of the CoGAPS algorithm, but I left a few comments. Hopefully, it helps in some small way!

File: R/methods-CogapsResult.R [hunk 1]
```diff
@@ -426,20 +426,21 @@ function(object, threshold, lp, axis)
...
+
         # rescale A in accordance with p's having max 1
         As <- sweep(As, 2, pscale, FUN="*")
-        
+
         # find the A with the highest magnitude
         Arowmax <- t(apply(As, 1, function(x) x/max(x)))
-        pmax <- apply(As, 1, max)
-        
+
```
[the remark this answers is not in the document]
Comment (dimalvovs) [reply R/methods-CogapsResult.R:443]: Thanks, to the point! The disagreement between the dimensions and hence inability to do the multiplication on the fly also bothered me. Maybe keeping the A and P discordant in the output was part of the masterplan, or may be it was a bug, but brutally changing output to agreeable dimensions looked too brave as it could break something downstream. So I started out with [tests/testthat/test_output_across_modes.R](https://github.com/FertigLab/CoGAPS/pull/103/files/186dce080fc597867c9adf0f05a8079843fbe344#diff-58622606deca119f6d5b90ad58782f3623e2481d60607176a8473b3e258c3496) to confirm that outputs disagree for all the modes, and confirmed with @jeanettejohnson that we historically let the user orient the matrices.

[the remark this answers is not in the document]
Comment (drbergman) [reply R/methods-CogapsResult.R:443]: ok, so the $P$ matrix here would be $n\times k$ and this all does make sense?

[the remark this answers is not in the document]
Comment (dimalvovs) [reply R/methods-CogapsResult.R:443]: yes, to some historic reason (GIST data test run shown with 7 patterns), the output is 
```
> dim(res@featureLoadings)
[1] 1363    7
> dim(res@sampleFactors)
[1] 9 7
```

File: R/class-CogapsResult.R
Comment (genesofeve) [inline R/class-CogapsResult.R:1]: please reinstate
